msal/public_client_application.go

```diff
@@ -70,13 +85,35 @@ func (pca *PublicClientApplication) AcquireTokenByDeviceCode(
 
 // AcquireTokenByAuthCode is a request to acquire a security token from the authority, using an authorization code.
 // Users need to create an AcquireTokenAuthCodeParameters instance and pass it in.
-func (pca *PublicClientApplication) AcquireTokenByAuthCode(
-	authCodeParams *AcquireTokenAuthCodeParameters) (AuthenticationResultProvider, error) {
+func (pca *PublicClientApplication) AcquireTokenByAuthCode(ctx context.Context, scopes []string, redirectURI string, options *AcquireTokenByAuthCodeOptions) (*msalbase.AuthenticationResult, error) {
+	authCodeParams := createAcquireTokenAuthCodeParameters(scopes, redirectURI)
 	authCodeParams.requestType = requests.AuthCodePublic
+	if options != nil {
+		authCodeParams.Code = options.Code
+		authCodeParams.CodeChallenge = options.CodeChallenge
+	}
 	return pca.clientApplication.acquireTokenByAuthCode(authCodeParams)
 }
 
-// GetAccounts gets all the accounts in the token cache.
-func (pca *PublicClientApplication) GetAccounts() []AccountProvider {
+// Accounts gets all the accounts in the token cache.
+// If there are no accounts in the cache the returned slice is empty.
+func (pca *PublicClientApplication) Accounts() []*msalbase.Account {
 	return pca.clientApplication.getAccounts()
 }
+
+// AcquireTokenSilentOptions contains the optional parameters to acquire a token silently (from cache).
+type AcquireTokenSilentOptions struct {
+	// Account specifies the account to use when acquiring a token from the cache.
+	Account *msalbase.Account
+}
+
+// AcquireTokenByDeviceCodeOptions contains the optional parameters used to acquire an access token using the device code flow.
+type AcquireTokenByDeviceCodeOptions struct {
```

Comment:
We use this in our track 2 SDKs to reduce the chance of breaking changes.  It works well for generated code where there tends to be more churn.  For hand-written APIs where we can be tactical it might not make as much sense.  Any thoughts on this one?

Reply:
I think it's fine to have an option object for all flows, though it there is nothing today then likelihood of anything in the future is relatively small. I would though actually have expected some option

---
In reply to: [516285138](https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/68#discussion_r516285138) [](ancestors = 516285138)